osf/models/files.py

```diff
@@ -1,273 +1,144 @@
         contains all the helpers required.
         A FileNode wraps a StoredFileNode to provider usable abstraction layer
     """
-
     # TODO DELETE ME POST MIGRATION
     modm_model_path = 'website.files.models.base.StoredFileNode'
     modm_query = None
     migration_page_size = 10000
-    # /TODO DELETE ME POST MIGRATION
+    # /TODO DELETE ME POST MIGRATION]
+    version_identifier = 'revision'  # For backwards compatibility
+    FOLDER, FILE, ANY = 0, 1, 2
 
+    # The User that has this file "checked out"
+    # Should only be used for OsfStorage
+    checkout = models.ForeignKey('osf.OSFUser', blank=True, null=True)
     # The last time the touch method was called on this FileNode
     last_touched = NonNaiveDateTimeField(null=True, blank=True)
     # A list of dictionaries sorted by the 'modified' key
     # The raw output of the metadata request deduped by etag
     # Add regardless it can be pinned to a version or not
-    history = DateTimeAwareJSONField(default=[], blank=True)
+    _history = DateTimeAwareJSONField(default=[], blank=True)
     # A concrete version of a FileNode, must have an identifier
     versions = models.ManyToManyField('FileVersion')
 
-    node = models.ForeignKey('AbstractNode', blank=True, null=True)
-    parent = models.ForeignKey('StoredFileNode', blank=True, null=True, default=None, related_name='child')
-    copied_from = models.ForeignKey('StoredFileNode', blank=True, null=True, default=None,
-                                    related_name='copy_of')
+    node = models.ForeignKey('osf.AbstractNode', blank=True, null=True, related_name='files')
+    parent = models.ForeignKey('self', blank=True, null=True, default=None, related_name='_children')
+    copied_from = models.ForeignKey('self', blank=True, null=True, default=None, related_name='copy_of')
 
-    trashed_children = GenericRelation('TrashedFileNode')
-
-    is_file = models.BooleanField(default=True)
     provider = models.CharField(max_length=25, blank=False, null=False, db_index=True)
 
-    name = models.CharField(max_length=1000, blank=True, null=True)
-    path = models.CharField(max_length=2000, blank=True, null=True)  # 1950 on prod
-    _materialized_path = models.CharField(max_length=1000, blank=True, null=True)  # 482 on staging
+    name = models.TextField(blank=True, null=True)
+    _path = models.TextField(blank=True, null=True)  # 1950 on prod
+    _materialized_path = models.TextField(blank=True, null=True)  # 482 on staging
 
-    # The User that has this file "checked out"
-    # Should only be used for OsfStorage
-    checkout = models.ForeignKey('OSFUser', blank=True, null=True)
+    is_deleted = False
+    deleted_on = NonNaiveDateTimeField(blank=True, null=True)
+    deleted_by = models.ForeignKey('osf.OSFUser', related_name='files_deleted_by', null=True, blank=True)
+
+    objects = BaseFileNodeManager()
+
+    class Meta:
+        index_together = (
+            ('node', 'type', 'provider', '_path'),
```

Comment:
I remember us talking about this but no, off the top of my head I don't remember why.